lib/galaxy/tool_util/linters/general.py

```diff
@@ -63,8 +63,12 @@ def lint_general(tool_source, lint_ctx):
 
     requirements, containers = tool_source.parse_requirements_and_containers()
     for r in requirements:
+        if r.name == '':
```

Comment:
Maybe constraint these checks to package requirements?

Reply:
And we should also lint the `type` attribute, I think.

Reply:
> Maybe constraint these checks to package requirements?

True.

> And we should also lint the type attribute, I think.

I think this should be covered by the xsd https://github.com/galaxyproject/galaxy/blob/b793ccdb463fef14e54a62fe8430807b96e8875b/lib/galaxy/tool_util/xsd/galaxy.xsd#L6076

But then I do not know why missing versions are not caught by the xsd check (https://github.com/galaxyproject/galaxy/blob/b793ccdb463fef14e54a62fe8430807b96e8875b/lib/galaxy/tool_util/xsd/galaxy.xsd#L671)

Reply:
> I think this should be covered by the xsd 

Correct, sorry for the noise!